ingest-app/src/cmr/ingest/runner.clj

```diff
@@ -1,16 +1,14 @@
 (ns cmr.ingest.runner
   "Entry point for the application. Defines a main method that accepts arguments."
   (:require [cmr.ingest.system :as system]
-            [clojure.string :as string]
-            [cmr.common.log :as log :refer (debug info warn error)]
-            [cmr.common.api.web-server :as web]
-            [cmr.ingest.api.routes :as routes]
+            [cmr.common-app.app-events :as events]
             [cmr.common.config :as cfg])
   (:gen-class))
 
 (defn -main
   "Starts the App."
   [& args]
   (let [system (system/start (system/create-system))]
-    (info "Running...")
-    (cfg/check-env-vars)))
+    (cfg/check-env-vars)
+    (events/stop-on-exit-hook (:instance-name system) #(system/stop system))
+    (events/dump-on-exit-hook (:instance-name system))))
```

Comment:
So does dump-on-exit get invoked (and the dump file created if configured to do so) for _all_  shutdowns? Hm. Maybe I should take this question up outside of the PR.

Reply:
a kill -9 never gets here, that is done at the OS level, a kill -15 and control-c (local) show up here.

Reply:
Yeah, I understand. My question is one of how to manage the all of the dump files we don't care about (nominal shutdown) vs the ones we do care about (AWS Fargate sends the SIGTERM prior to the SIGKILL). The real value of this ticket and functionality is that we capture the stack trace at the time of task failure, during Fargate initiated termination. We need to be able to separate the wheat from the chaff, since the task (and we have 7 of them for Search and Ingest) will create this file everytime we deploy.
This needs to be taken up as part of the next ticket, which notionally is simply "writing the file to some accessible place" but really also needs to be identification and management so that we don't overwrite the files we want to preserve for investigation with those which are just generated as part of the normal shutdown process.

Reply:
I don't have a way to do that. The process to turn on the dump is an ENV variable that is only settable at deploy time, however, I could check some other resource or provide an end point you could hit to toggle the behavior on. In fact, I don't see why that would be a problem, to write an end point (like clear cache) that you hit when you want to start capturing dumps.